File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/data/daos/impl/DeviceInfoDao.java
package com.dkellycollins.triggerfinger.data.daos.impl;

import android.view.View;

import com.dkellycollins.triggerfinger.data.daos.IDeviceInfoDao;
import com.dkellycollins.triggerfinger.util.qa.Assert;

public class DeviceInfoDao implements IDeviceInfoDao {

    private final View _view;

    public DeviceInfoDao(View view) {
        Assert.isNotNull(view, "view");

        _view = view;
    }

    @Override
    public int getWidth() {
        return _view.getWidth();
    }

    @Override
    public int getHeight() {
        return _view.getHeight();
    }
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/data/daos/impl/DeviceInfoDao.kt
package com.dkellycollins.triggerfinger.data.daos.impl

import android.view.View

import com.dkellycollins.triggerfinger.data.daos.IDeviceInfoDao

class DeviceInfoDao(private val _view: View) : IDeviceInfoDao {

    override val width: Int
        get() = _view.width

    override val height: Int
        get() = _view.height
}
